addons/crm/models/crm_team.py

```diff
@@ -565,6 +565,13 @@ def _allocate_leads_deduplicate(self, leads, duplicates_cache=None):
             'duplicates': leads_dup_ids,
         }
 
+    def _get_lead_to_assign_domain(self, teams):
```

Comment:
As you are already on team model, could be a classic multi method (team_id, in, self.ids)